### PR TITLE
fix failing interfaces

### LINKED ISSUE #11683: EVM Engineering: fix broken interfaces
Some existing contract interfaces don't entirely cover the contract ABI for the underlying contract. Such interfaces are revealed by #11677. We should fix those interfaces so they can be used as full drop-in replacements for the contract(s).

### PR DESCRIPTION
**Pull request by AmadiMichael**
Fixes https://github.com/ethereum-optimism/optimism/issues/11683

### CODE REVIEW

File: packages/contracts-bedrock/src/L2/CrossL2Inbox.sol [hunk 1]
```diff
@@ -65,13 +65,13 @@ contract CrossL2Inbox is ICrossL2Inbox, ISemver, TransientReentrancyAware {
...
-    /// @custom:semver 1.0.0-beta.8
-    string public constant version = "1.0.0-beta.8";
+    /// @custom:semver 1.0.0-beta.9
+    string public constant version = "1.0.0-beta.9";
 
     /// @notice Emitted when a cross chain message is being executed.
     /// @param msgHash Hash of message payload being executed.
     /// @param id Encoded Identifier of the message.
-    event ExecutingMessage(bytes32 indexed msgHash, Identifier id);
+    event ExecutingMessage(bytes32 indexed msgHash, ICrossL2Inbox.Identifier id);
```
Comment (smartcontracts) [inline packages/contracts-bedrock/src/L2/CrossL2Inbox.sol]: We shouldn't import the interface within this file, instead we should re-define the struct within this contract.

File: packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol [hunk 2]
```diff
@@ -87,6 +87,8 @@ contract FaultDisputeGame is Clone, ISemver {
         address counteredBy;
     }
 
+    event Initialized(uint8 version);
```
Comment (smartcontracts) [inline packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol]: We should be able to remove this as long as we keep `IInitializable` in the ignore list, right? I don't like defining an event here that doesn't get used.

Comment (AmadiMichael) [reply packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol]: Yeah

File: packages/contracts-bedrock/src/dispute/interfaces/IInitializable.sol [hunk 3]
```diff
@@ -2,5 +2,5 @@
 pragma solidity ^0.8.0;
 
 interface IInitializable {
-    function initialize() external payable;
+    event Initialized(uint8 version);
```
Comment (smartcontracts) [inline packages/contracts-bedrock/src/dispute/interfaces/IInitializable.sol]: Can revert this change

File: packages/contracts-bedrock/src/dispute/interfaces/IDisputeGame.sol [hunk 4]
```diff
@@ -7,6 +7,8 @@ import "src/dispute/lib/Types.sol";
 interface IDisputeGame is IInitializable {
     event Resolved(GameStatus indexed status);
 
+    function initialize() external payable;
```
Comment (smartcontracts) [inline packages/contracts-bedrock/src/dispute/interfaces/IDisputeGame.sol]: Can revert this change

File: packages/contracts-bedrock/src/universal/interfaces/IOptimismMintableERC20.sol [hunk 5]
```diff
@@ -1,31 +1,80 @@
...
 
-import { IERC165 } from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import { ISemver } from "src/universal/interfaces/ISemver.sol";
 
 /// @title IOptimismMintableERC20
 /// @notice This interface is available on the OptimismMintableERC20 contract.
 ///         We declare it as a separate interface so that it can be used in
 ///         custom implementations of OptimismMintableERC20.
-interface IOptimismMintableERC20 is IERC165 {
+interface IOptimismMintableERC20 is ISemver {
```
Comment (smartcontracts) [inline packages/contracts-bedrock/src/universal/interfaces/IOptimismMintableERC20.sol]: Need to revert back to original (broken) interface for now

File: packages/contracts-bedrock/src/universal/interfaces/IOptimismMintableERC20.sol [hunk 6]
```diff
@@ -1,31 +1,80 @@
...
 
     function burn(address _from, uint256 _amount) external;
-}
 
-/// @custom:legacy
-/// @title ILegacyMintableERC20
-/// @notice This interface was available on the legacy L2StandardERC20 contract.
-///         It remains available on the OptimismMintableERC20 contract for
-///         backwards compatibility.
-interface ILegacyMintableERC20 is IERC165 {
```
Comment (smartcontracts) [inline packages/contracts-bedrock/src/universal/interfaces/IOptimismMintableERC20.sol:25]: Same as above, need to revert to original broken interface but let's still put this in a separate file

**Comment by smartcontracts** [comment]
All of the semgrep failures can be added to the ignore list

File: packages/contracts-bedrock/src/L2/interfaces/ICrossL2Inbox.sol [hunk 7]
```diff
@@ -1,53 +1,70 @@
...
-    /// @return timestamp_ The timestamp of the Identifier.
-    function timestamp() external view returns (uint256 timestamp_);
+    function timestamp() external view returns (uint256);
 
     /// @notice Returns the chain ID of the Identifier.
-    /// @return chainId_ The chain ID of the Identifier.
-    function chainId() external view returns (uint256 chainId_);
+    function chainId() external view returns (uint256);
+
+    function setInteropStart() external;
```
Comment (ControlCplusControlV) [inline packages/contracts-bedrock/src/L2/interfaces/ICrossL2Inbox.sol:61]: no commets / natspec here? Or why does `function setInteropStart() external` look so out of place

Comment (AmadiMichael) [reply packages/contracts-bedrock/src/L2/interfaces/ICrossL2Inbox.sol:61]: The natspec are only required in the contract itself. Also the intention in the near future is for interfaces to be auto generated

**Review by smartcontracts — APPROVED** [review]
